File: 005/sol.cpp
#include <iostream>

int main() {
    const int n = 20;
    int count;
    int i;

    for (i = n; ; i += n) {
        count = 0;
        for (int j = 1; j <= n; j++) {
            if (i % j == 0) count++;
        }
        if (count == n) break;
    }
    std::cout << i;
}

File: 092/sol.py
def chain(n):
    while True:
        n = sum(int(d)**2 for d in str(n))
        if n == 1 or n == 89:
            break
    return n

def solve():
    count = 0
    for i in range(1, 10**7):
        if chain(i) == 89:
            count += 1
    return count 

if __name__ == "__main__":
    # takes about 1 minute
    print(solve())
